src/pymmcore_plus/mda/metadata/schema.py

```diff
@@ -0,0 +1,169 @@
+from typing import Any, Dict, List, Literal, Optional, Tuple, TypedDict
+
+import useq
+from typing_extensions import NotRequired
+
+__all__ = [
+    "FrameMetaV1",
+    "SummaryMetaV1",
+    "ConfigGroup",
+    "ConfigPreset",
+    "DeviceInfo",
+    "ImageInfo",
+    "PixelSizeConfigPreset",
+    "Position",
+    "PropertyInfo",
+    "PropertyValue",
+    "SystemInfo",
+]
+
+
+class PropertyInfo(TypedDict):
+    """Information about a single device property."""
+
+    name: str
+    value: Optional[str]
+    data_type: Literal["undefined", "float", "int", "str"]
+    allowed_values: Optional[Tuple[str, ...]]
+    is_read_only: bool
+    is_pre_init: bool
+    limits: NotRequired[Tuple[float, float]]
+    sequenceable: bool
+    sequence_max_length: NotRequired[int]
+    # device_label: str
+
+
+class DeviceInfo(TypedDict):
+    """Information about a specific device."""
+
+    label: str
+    library: str
+    name: str
+    type: str
+    description: str
+    properties: Tuple[PropertyInfo, ...]
+    parent_label: Optional[str]  # None on hub devices
+
+    # state device only
+    labels: NotRequired[Tuple[str, ...]]
+    # focus device only
+    focus_direction: NotRequired[Literal["Unknown", "TowardSample", "AwayFromSample"]]
+    # hub device only
+    child_names: NotRequired[Tuple[str, ...]]
+    # stage or XYstage device only
+    is_stage_sequenceable: NotRequired[bool]
+    # stage device only
+    is_continuous_focus_drive: NotRequired[bool]
+    is_stage_linear_sequenceable: NotRequired[bool]
+    # camera device only
+    is_exposure_sequenceable: NotRequired[bool]
+
+
+class SystemInfo(TypedDict):
+    """General system information."""
+
+    pymmcore_version: str
+    pymmcore_plus_version: str
+    mmcore_version: str
+    device_api_version: str
+    device_adapter_search_paths: Tuple[str, ...]
+    system_configuration: Optional[str]
+    primary_log_file: str
+    circular_buffer_memory_footprint: int
+    buffer_total_capacity: int
+    buffer_free_capacity: int
+    timeout_ms: int
+    # remaining_image_count: int
+    continuous_focus_enabled: bool
+    continuous_focus_locked: bool
+    auto_shutter: bool
+
+
+class ImageInfo(TypedDict):
+    """Information about the current image structure."""
+
+    bytes_per_pixel: int
```

Comment:
I would consider adding a field that specifies the general category of pixel format, with options like "gray" and "ABGR". The main reason is to allow readers to reject any future pixel formats that it does not know how to interpret. Such future formats may include RGB (without the wasted channel), YUV (perhaps), or RAW (with a new field specifying the Bayer mask type).

Fully general pixel format specs (cf. GenICam Pixel Format Naming Convention) would be overkill, but by having even one category (could even be called "mm-classic", combining gray and ABGR), future extensions can be made without breaking readers.

Also, it is worth considering using bytes-per-component rather than bytes-per-pixel.